### PR TITLE
[MRG+1] Region volume

### LINKED ISSUE #1446: region_extractor.connected_regions gets zero volume from affine
in 
https://github.com/nilearn/nilearn/blob/master/nilearn/regions/region_extractor.py#L188

volume is computed from affine diagonal. This is making my extraction fails because my affine is
```Python
np.array([[0.3, 0   , 0  , -14.55],
          [0  , 0   , 0.3, -14.55],
          [0  , -0.3, 0  ,  14.55],
          [0  , 0   , 0  ,  1]])
```

What do you think about getting the volume information from `header.get_zooms()`

### PR DESCRIPTION
**Pull request by salma1601**
try fix #1446

### CODE REVIEW

File: nilearn/regions/tests/test_region_extractor.py [hunk 1]
```diff
@@ -185,6 +185,18 @@ def test_region_extractor_fit_and_transform():
     assert_not_equal(extract_ratio.regions_img_, '')
     assert_true(extract_ratio.regions_img_.shape[-1] >= 9)
 
+    # smoke test with permuted axes
+    affine = np.eye(4)
+    affine[[0, 1]] = affine[[1, 0]]
```
Comment (bthirion) [inline nilearn/regions/tests/test_region_extractor.py]: I don't understand; this line does not do anything.

File: nilearn/regions/tests/test_region_extractor.py [hunk 2]
```diff
@@ -185,6 +185,18 @@ def test_region_extractor_fit_and_transform():
...
+    maps, mask_img = generate_maps((40, 40, 40), n_regions=n_regions,
+                                   affine=affine)
+
+    extract_ratio = RegionExtractor(maps, threshold=0.2,
+                                    thresholding_strategy='ratio_n_voxels')
+    extract_ratio.fit()
+    assert_not_equal(extract_ratio.regions_img_, '')
+    assert_true(extract_ratio.regions_img_.shape[-1] >= 9)
```
Comment (bthirion) [inline nilearn/regions/tests/test_region_extractor.py]: It is unclear to me why this new test enforces the changes that you added. Can you explain a bit more in a comment ?

**Comment by bthirion** [comment]
Thx for the correction. I have 2 comments.

File: nilearn/regions/tests/test_region_extractor.py (hunk 2 above)
Comment (salma1601) [reply nilearn/regions/tests/test_region_extractor.py]: it permutes the first and second lines
```Python
In [22]: a = np.eye(4)

In [23]: a[[0, 1]] = a[[1, 0]]

In [24]: a
Out[24]: 
array([[ 0.,  1.,  0.,  0.],
       [ 1.,  0.,  0.,  0.],
       [ 0.,  0.,  1.,  0.],
       [ 0.,  0.,  0.,  1.]])
```

Comment (bthirion) [reply nilearn/regions/tests/test_region_extractor.py]: Sorry for misunderstanding.
LGTM then

**Comment by salma1601** [comment]
I have test failures that seem unrelated to my changes

**Comment by KamalakerDadi** [comment]
I see examples are good as before with this patch. +1 merge.
Can you please rebase ?

**Comment by salma1601** [comment]
I rebased. But there is still one travis failiure with python 3.5 for `test_canica.test_canica_square_img`
`test_dict_learning.test_dict_learning` but none of the functions used in there rely on region extraction AFAIK so I don't understand

**Comment by KamalakerDadi** [comment]
>But there is still one travis failiure with python 3.5 for test_canica.test_canica_square_img
test_dict_learning.test_dict_learning but none of the functions used in there rely on region extraction AFAIK so I don't understand

I am not totally correct. But, I think that there might be issues related to recent scipy. I will have a look.

**Comment by KamalakerDadi** [comment]
>    But there is still one travis failiure with python 3.5 for test_canica.test_canica_square_img
 >   test_dict_learning.test_dict_learning but none of the functions used in there rely on region extraction >AFAIK so I don't understand

>I am not totally correct. But, I think that there might be issues related to recent scipy. I will have a look.

The failures are related to SciPy and they are fixed now. Could please rebase again ?
If everything goes green. +1 for merge

**Comment by GaelVaroquaux** [comment]
LGTM. Merging! Thx

**Pull request merged**